src/Nix/Utils.hs

```diff
@@ -108,3 +109,17 @@ toEncodingSorted = \case
                 $ M.toList m
     A.Array l -> A.list toEncodingSorted $ V.toList l
     v -> A.toEncoding v
+
+data NixPathEntryType = PathEntryPath | PathEntryURI deriving (Show, Eq)
```

Comment:
I don't think `Nix.Utils` is the place for this type to live... And should we just name is `NixPath`?